tests/model.py

```diff
@@ -38,40 +38,38 @@ def tearDown(self):
-        submodel.sp2 = Parameter([2, 4], I.Constant(0))
-        submodel.sp2.value = tF.input(np.array([[9,8,7,6],[5,4,3,2]]))
-        submodel.add("sp1", submodel.sp1)
-        submodel.add("sp2", submodel.sp2)
+        sp1 = Parameter([2, 4], I.Constant(0))
+        sp1.value = tF.input(np.array([[0,1,2,3],[4,5,6,7]]))
+        sp2 = Parameter([2, 4], I.Constant(0))
+        sp2.value = tF.input(np.array([[9,8,7,6],[5,4,3,2]]))
+        submodel.add("sp1", sp1)
+        submodel.add("sp2", sp2)
         parentmodel = TestModel()
-        parentmodel.p1 = Parameter([4, 2], I.Constant(0))
-        parentmodel.p1.value = tF.input(np.array([[0,1],[2,3],[4,5],[6,7]]))
-        parentmodel.p2 = Parameter([4, 2], I.Constant(0))
-        parentmodel.p2.value = tF.input(np.array([[9,8],[7,6],[5,4],[3,2]]))
-        parentmodel.sub = submodel
-        parentmodel.add("p1", parentmodel.p1)
-        parentmodel.add("p2", parentmodel.p2)
-        parentmodel.add("sub", parentmodel.sub)
+        p1 = Parameter([4, 2], I.Constant(0))
+        p1.value = tF.input(np.array([[0,1],[2,3],[4,5],[6,7]]))
+        p2 = Parameter([4, 2], I.Constant(0))
+        p2.value = tF.input(np.array([[9,8],[7,6],[5,4],[3,2]]))
+        parentmodel.add("p1", p1)
+        parentmodel.add("p2", p2)
+        parentmodel.add("sub", submodel)
         submodel_load = TestModel()
-        submodel_load.sp1 = Parameter()
-        submodel_load.sp2 = Parameter()
-        submodel_load.add("sp1", submodel_load.sp1)
-        submodel_load.add("sp2", submodel_load.sp2)
+        sp1 = Parameter()
+        sp2 = Parameter()
+        submodel_load.add("sp1", sp1)
+        submodel_load.add("sp2", sp2)
         parentmodel_load = TestModel()
-        parentmodel_load.p1 = Parameter()
-        parentmodel_load.p2 = Parameter()
-        parentmodel_load.sub = submodel_load
-        parentmodel_load.add("p1", parentmodel_load.p1)
-        parentmodel_load.add("p2", parentmodel_load.p2)
-        parentmodel_load.add("sub", parentmodel_load.sub)
+        p1 = Parameter()
+        p2 = Parameter()
+        parentmodel_load.add("p1", p1)
+        parentmodel_load.add("p2", p2)
+        parentmodel_load.add("sub", submodel_load)
         with tempfile.NamedTemporaryFile() as fp:
             parentmodel.save(fp.name)
             parentmodel_load.load(fp.name)
-        self.assertTrue((parentmodel_load.p1.value.to_ndarrays()[0] == np.array([[0,1],[2,3],[4,5],[6,7]])).all())
-        self.assertTrue((parentmodel_load.p2.value.to_ndarrays()[0] == np.array([[9,8],[7,6],[5,4],[3,2]])).all())
-        self.assertTrue((parentmodel_load.sub.sp1.value.to_ndarrays()[0] == np.array([[0,1,2,3],[4,5,6,7]])).all())
-        self.assertTrue((parentmodel_load.sub.sp2.value.to_ndarrays()[0] == np.array([[9,8,7,6],[5,4,3,2]])).all())
+        self.assertTrue((parentmodel_load["p1"].value.to_ndarrays()[0] == np.array([[0,1],[2,3],[4,5],[6,7]])).all())
```

Comment:
Why had the behavior been changed?

Reply:
This test registers parameters and models without using `__setattr__`, so we cannot use `__getattr__` (like `parentmodel_load.p1`) to retrieve parameter objects.

Reply:
Okay. It should be documented in `Model.add()` and `Model.__setattr__()` because some users may expect that `add()` also adds corresponding fields.